### ISSUE TITLE
A few more demangler fails

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#410]
Please see below. Thanks!
```
demangle: simple demangle failed:
_ZTWN6__xray19__xray_fdr_internal7RunningE
  ^
dd_special_name:1169: "valid special name" expected
demangle: current:  (pos: 2/42)
demangle:   [0] dd_encoding
```
```
demangle: simple demangle failed:
_ZTW21__sancov_lowest_stack
  ^
dd_special_name:1169: "valid special name" expected
demangle: current:  (pos: 2/27)
demangle:   [0] dd_encoding
```

**Comment by honggyukim** [comment]
They are interpreted as TLS wrapper function by the full demangler.
```
$ ./misc/demangler --full _ZTWN6__xray19__xray_fdr_internal7RunningE
TLS wrapper function for __xray::__xray_fdr_internal::Running

$ ./misc/demangler --full _ZTW21__sancov_lowest_stack
TLS wrapper function for __sancov_lowest_stack
```

**Comment by namhyung** [comment]
Here's the fix:

```diff
diff --git a/utils/demangle.c b/utils/demangle.c
index 7ca3bb1a..777603a6 100644
--- a/utils/demangle.c
+++ b/utils/demangle.c
@@ -1130,6 +1130,24 @@ static int dd_special_name(struct demangle_data *dd)
                                return -1;
                        return dd_encoding(dd);
                }
+               if (c1 == 'C') {
+                       /* construction vtable */
+                       dd_consume_n(dd, 2);
+                       if (dd_type(dd) < 0)
+                               return -1;
+                       if (dd_number(dd) < 0)
+                               return -1;
+                       return dd_type(dd);
+               }
+               if (c1 == 'H' || c1 == 'W') {
+                       /* TLS init and wrapper */
+                       dd_consume_n(dd, 2);
+                       if (dd->newpos)
+                               dd_append(dd, "::");
+                       dd_append(dd, "TLS_");
+                       dd_append(dd, c1 == 'H' ? "init" : "wrap");
+                       return dd_name(dd);
+               }
        }
        if (c0 == 'G') {
                if (c1 == 'V') {
```

Testing..

```
$ misc/demangler _ZTWN6__xray19__xray_fdr_internal7RunningE
TLS_wrap::__xray::__xray_fdr_internal::Running
```

**Comment by honggyukim** [comment]
It works fine. Thanks for the fix!

**Comment by namhyung** [comment]
Thank you for testing.  Anyway I'd not call it as a bug since it didn't do anything *wrong* by itself.

**Comment by honggyukim** [comment]
I see. I will put `bug` tag only if segfault like crash is shown.

**Comment by namhyung** [comment]
Merged: 1f00a036
